File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class CategoryBase(BaseModel):
    name: str
    slug: str
    description: Optional[str] = None
    image_url: Optional[str] = None
    parent_id: Optional[int] = None


class CategoryCreate(CategoryBase):
    pass


class CategoryOut(CategoryBase):
    id: int
    created_at: datetime
    children: list["CategoryOut"] = []

    class Config:
        from_attributes = True


class ProductBase(BaseModel):
    name: str
    slug: str
    description: Optional[str] = None
    short_description: Optional[str] = None
    price: Optional[float] = None
    regular_price: Optional[float] = None
    sale_price: Optional[float] = None
    sku: Optional[str] = None
    stock: int = 0
    in_stock: bool = True
    image_url: Optional[str] = None
    category_id: Optional[int] = None
    featured: bool = False


class ProductCreate(ProductBase):
    pass


class ProductOut(ProductBase):
    id: int
    category: Optional[CategoryOut] = None
    created_at: datetime

    class Config:
        from_attributes = True


class ProductListOut(BaseModel):
    items: list[ProductOut]
    total: int
    page: int
    pages: int

File: frontend/src/domain/entities/Product.ts
export interface Category {
  id: number;
  name: string;
  slug: string;
  description?: string;
  image_url?: string;
  parent_id?: number;
  created_at: string;
  children?: Category[];
}

export interface Product {
  id: number;
  name: string;
  slug: string;
  description?: string;
  short_description?: string;
  price?: number;
  regular_price?: number;
  sale_price?: number;
  sku?: string;
  stock?: number;
  in_stock: boolean;
  image_url?: string;
  category?: Category;
  featured: boolean;
  meta_title?: string | null;
  meta_description?: string | null;
  created_at: string;
}

export interface ProductList {
  items: Product[];
  total: number;
  page: number;
  pages: number;
}

export interface GetProductsParams {
  page?: number;
  per_page?: number;
  category?: string;
  search?: string;
  featured?: boolean;
}
